### PR TITLE
Split libraries into independent Gradle projects

### PR DESCRIPTION
**Pull request by thunderbiscuit**
## Description
This PR refactors the repository into two directories, each containing separate libraries as completely independent Gradle builds/projects.

Note that in order to be really careful I fixed the CI publishing workflow but brought it back to simply publishing to the staging repository (it's easy to make a typo and send the compiled binaries to the wrong directory) so that we can check them and see that everything is in the right place before publishing if we trigger the workflow and everything works well we can just make a quick fix that does the full publishing and merge that on the day of the release.

### Notes to the reviewers
This is a big refactor so I want to make sure I think of everything:
- [x] Readme
- [x] Custom plugins build both libraries correctly
- [x] Build scripts all work
- [x] CI
- [x] Other "community" files

### Checklists
* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
Question for you @notmandatory: do you think the license files should be added to each of the directories? It feels that way to me, just so there is no mistaking it (even though it's sort of obvious).

**Review by notmandatory — APPROVED** [review]
ACK 5450c084a68aea848d328a81f93bdb68a670c108

I found one very small language nit.. but other than that looks good. I locally tested the android and jvm build and test maven tasks on my Mac M1.

**Comment by notmandatory** [comment]
Looks like you also may need to do a rebase.

File: bdk-jvm/plugins/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Readme
+The purpose of this directory is to host the Gradle plugin that add tasks for building the native binaries required by bdk-jvm, and building the language bindings files.
```
[the remark this answers is not in the document]
Comment (thunderbiscuit) [reply bdk-jvm/plugins/README.md:2]: Good catch!

**Pull request merged**